src/HexViewer.ts

```diff
@@ -100,11 +100,11 @@ export class HexViewer {
     private content: JQuery;
     private contentOuter: JQuery;
 
-    public mouseDownOffset: number;
     private canDeselect: boolean;
-
     public selectionStart: number = -1;
     public selectionEnd: number = -1;
+    public selectionDragStart: number;
+    public selectionDragEnd: number = 0;
```

Comment:
To me, these new two properties seem unnecessary, as they only reflect the values in `selectionStart` and `selectionEnd`. The only difference is that always applies `selectionStart <= selectionEnd`, but about your new `*Drag*` properties cannot be said that. So only two cases can happen (when we don't count the case `selectionStart == selectionEnd`, but then are all 4 properties equal of course, so we don't have to care):

Property \ case | `selectionDragStart < selectionDragEnd` | `selectionDragStart > selectionDragEnd`
------------ | ------------ | -------------
`selectionDragStart ==`  | `selectionStart` | `selectionEnd`
`selectionDragEnd ==`  | `selectionEnd` | `selectionStart`

So if you need to differentiate between the two cases, it would be enough to create a `boolean` property, called `isSelectionLeftToRight` for example (while `isSelectionLeftToRight == selectionDragStart < selectionDragEnd`). Then applies `selectionDragStart == isSelectionLeftToRight ? selectionStart : selectionEnd`, so you can substitute that.

I know, this expresssion may seem more verbose, but I actually think that the code will be a bit shorter and more understandable as a result. And we get rid of two `number` properties which carry a very little information compared to their range of values, and store only the data we need. It's generally best to avoid duplicate data, because if you're not careful and update just one pair of the properties, you'll have inconsistent data in your app and that's likely to cause weird bugs somewhere.